### PR TITLE
Update Lambda wrapper to allow omission of input/output fields

### PR DESCRIPTION
**Pull request by fitzoh**
Currently the Lambda wrapper always records the incoming event and the lambda response (if present).
That behavior may be undesirable if the input/output is very large.

This PR updates the Lambda wrapper to include two parameters (`record_input` and `record_output`) which allow the customization of that behavior.  These parameters default to `True`, leaving the default behavior unchanged.

### CODE REVIEW

File: beeline/middleware/awslambda/__init__.py
Comment (fitzoh) [inline beeline/middleware/awslambda/__init__.py:157]: Open to suggestions on a better name

File: beeline/middleware/awslambda/test_awslambda.py
Comment (fitzoh) [inline beeline/middleware/awslambda/test_awslambda.py:176]: duplicated this test to make sure it works both called and uncalled (```@awslambda.beeline_wrapper``` vs ```@awslambda.beeline_wrapper()```)

File: beeline/middleware/awslambda/__init__.py
Comment (MikeGoldsmith) [reply beeline/middleware/awslambda/__init__.py:157]: I think this is fine, it's internal and is clear enough :+1:

File: beeline/middleware/awslambda/test_awslambda.py
Comment (MikeGoldsmith) [inline beeline/middleware/awslambda/test_awslambda.py:216]: Should this also set `record_output` to **False**? The assert below verifies `app.response` is not in the response object.

**Review by MikeGoldsmith — CHANGES_REQUESTED** [review]
This looks good, thanks for the contribution!

I've one query regarding the test case for not recording response.

File: beeline/middleware/awslambda/test_awslambda.py
Comment (fitzoh) [reply beeline/middleware/awslambda/test_awslambda.py:216]: Yeah, it should have... ended up just making separate tests for input/output in 593fe92b1bf5722d4daef5e59c6e1079a126dbb4

**Review by MikeGoldsmith — APPROVED** [review]
This LGTM - thanks!

@dsoo would you mind taking a look too please?

**Comment by MikeGoldsmith** [comment]
@paulosman please can you take a look as a second pair of eyes?

**Comment by MikeGoldsmith** [comment]
@fitzoh sorry fo the delay in getting this a second review. I'll get someone to look at this in the next day so we can get it signd off and added into the next release.

Thanks again.

**Comment by fitzoh** [comment]
No worries, thanks for following up on it

**Comment by MikeGoldsmith** [comment]
Closed & re-opened to trigger CI.

File: beeline/middleware/awslambda/test_awslambda.py
Comment (martin308) [inline beeline/middleware/awslambda/test_awslambda.py:177]: we might need to rename this to get the linter to pass the build

**Review by martin308 — CHANGES_REQUESTED** [review]
Looks good to me, looks like just one change required to get the linter to be happy

File: beeline/middleware/awslambda/test_awslambda.py
Comment (MikeGoldsmith) [inline beeline/middleware/awslambda/test_awslambda.py:180]: As @martin308 pointed out, we have a duplicate function name which the linter isn't happy with. The suggestion below should fix it.

```suggestion
            def bar(event, context):
                return 1

            self.assertEqual(bar(None, None), 1)
```

**Review by MikeGoldsmith — APPROVED** [review]

File: beeline/middleware/awslambda/test_awslambda.py
Comment (fitzoh) [reply beeline/middleware/awslambda/test_awslambda.py]: Updated in 6d16c2bc93acbc3ace6a21a39407358ff183d670 (thanks @MikeGoldsmith!)

Sorry for the delay, I've been a bit distracted the past week or so for obvious reasons 🙃

**Comment by MikeGoldsmith** [comment]
Thanks for fixing the linting issues @fitzoh - if you do one final merge / rebase from main, you'll get a recent change to our CI workflow that will allow it to pass.

**Comment by fitzoh** [comment]
>Thanks for fixing the linting issues @fitzoh 

Welcome

>if you do one final merge / rebase from main, you'll get a recent change to our CI workflow that will allow it to pass.

Done

**Review by martin308 — APPROVED** [review]

**Pull request merged**
